R/process_extract_cmh.R

```diff
@@ -0,0 +1,77 @@
+#' Process the CMH extract
+#'
+#' @description This will read and process the
+#' cmh extract, it will return the final data
+#' but also write this out as a zsav and rds.
+#'
+#' @param year The year to process, in FY format.
+#' @param data The extract to process
+#' @param write_to_disk (optional) Should the data be written to disk default is
+#' `TRUE` i.e. write the data to disk.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#' @family process extracts
+process_extract_cmh <- function(year, data, write_to_disk = TRUE) {
+  # Only run for a single year
+  stopifnot(length(year) == 1)
+
+  # Check that the supplied year is in the correct format
+  year <- check_year_format(year)
+
+  # Data Cleaning  ---------------------------------------
+
+  cmh_clean <- data %>%
+    # create recid, year, SMRType variables
+    dplyr::mutate(
+      recid = "CMH",
+      smrtype = add_smr_type(recid = .data$recid),
+      year = year
+    ) %>%
+    # contact end time
+    dplyr::mutate(
+      keyTime1 = lubridate::hms(keyTime1),
+      duration = lubridate::hms(duration)
+    ) %>%
+    dplyr::mutate(keyTime2 = .data$keyTime1 + .data$duration) %>%
```

Comment:
```suggestion
    dplyr::mutate(keyTime2 = hms::as.hms(.data$keyTime1 + lubridate::dminutes(.data$duration))) %>%
```
This solves the parsing issue by making sure `keyTime1` is a time of day, and the duration of `duration` is added on to it in minutes. Then it makes sure `keyTime2` is of `hms` class

Reply:
Great work, thanks for the fix! Tested and works perfectly. Will commit and then merge this into MainR. Thanks! :)